File: 20221118-Cache/java/app/src/test/java/CacheTest.java
import org.junit.jupiter.api.*;

import static org.junit.jupiter.api.Assertions.*;

class CacheTest {
  @Test
  void answer() {
    Cache cache = new Cache();

   assertEquals(50,cache.solution(3,
       new String[] {
           "Jeju", "Pangyo", "Seoul", "NewYork", "LA",
           "Jeju", "Pangyo", "Seoul", "NewYork", "LA"}));
    assertEquals(52,cache.solution(5,
        new String[] {
            "Jeju", "Pangyo", "Seoul", "NewYork", "LA","SanFrancisco",
            "Seoul", "Rome", "Paris", "Jeju", "NewYork", "Rome"}));


  }

  @Test
  void edgeCase() {
    Cache cache = new Cache();
    assertEquals(16,cache.solution(2,
        new String[] {
            "Jeju", "Pangyo", "NewYork", "newyork"}));
  }
}

File: 20221118-Cache/javaScript/Cache.test.js
import solution, { remove } from './Cache';

test('Cache', () => {
  // expect(solution(
  //   3,
  //   ['Jeju', 'Pangyo', 'Seoul', 'NewYork', 'LA', 'Jeju', 'Pangyo', 'Seoul', 'NewYork', 'LA'],
  // )).toBe(50);

  expect(solution(
    3,
    ['Jeju', 'Pangyo', 'Seoul', 'Jeju', 'Pangyo', 'Seoul', 'Jeju', 'Pangyo', 'Seoul'],
  )).toBe(21);
  expect(solution(
    2,
    ['Jeju', 'Pangyo', 'Seoul', 'NewYork', 'LA', 'SanFrancisco', 'Seoul', 'Rome', 'Paris', 'Jeju', 'NewYork', 'Rome'],
  )).toBe(60);
  expect(solution(
    5,
    ['Jeju', 'Pangyo', 'Seoul', 'NewYork', 'LA', 'SanFrancisco', 'Seoul', 'Rome', 'Paris', 'Jeju', 'NewYork', 'Rome'],
  )).toBe(52);
  expect(solution(
    0,
    ['Jeju', 'Pangyo', 'Seoul', 'NewYork', 'LA'],
  )).toBe(25);
  expect(solution(
    2,
    ['Jeju', 'Pangyo', 'NewYork', 'newyork'],
  )).toBe(16);
});

// test('filtered', () => {
//   expect(remove(['a', 'b', 'c'], 'b')).toStrictEqual(['a', 'c']);
//   expect(remove(['jeju', 'pangyo', 'seoul'], 'jeju')).toStrictEqual(['pangyo', 'seoul']);
// });
